lodash/index.d.ts

```diff
@@ -11202,7 +11202,7 @@ declare namespace _ {
     }
 
     //_.cloneDeepWith
-    type CloneDeepWithCustomizer<TValue, TResult> = (value: TValue) => TResult;
+    type CloneDeepWithCustomizer<TValue, TResult> = (value: TValue, key?: number|string, object?: any, stack?: any) => TResult;
```

Comment:
I don't have a problem making optional parameters required if there's a good case for it. Just seems like this is the right case to use optional parameters.

Want to get this in so we don't have to keep modifying the library source for TS compilation to succeed.

**TS Coding Standards**
http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#optional-parameters-in-callbacks
> This has a very specific meaning: the done callback might be invoked with 1 argument or might be invoked with 2 arguments.

**Lodash API**
https://lodash.com/docs/4.17.4#cloneWith
https://lodash.com/docs/4.17.4#cloneDeepWith
> The customizer is invoked with up to four arguments; (value [, index|key, object, stack]).

**Lodash Source, as of latest commit**
https://github.com/lodash/lodash/blob/165572022d58d3e39e928b33ff9b9589cc60b67b/dist/lodash.js#L2648 (line 2648)
> result = object ? customizer(value, key, object, stack) : customizer(value);